howto/register_external_algorithm.md

```diff
@@ -0,0 +1,697 @@
+Here is an example of the `./my_awesome_configs/model_manager/ext_sota.yaml` config file:
+
+```yaml
+defaults:
+  - default
+  - _self_
+
+disabled: False
+models: 
+  agent:
+    model_name: "${exp_name}"
+    description: "SOTA Agent in ${env.id} Environment"
+    tags: {}
+```
+
+#### Experiment Configs
+
+In the second file, you have to specify all the elements you want in your experiment and you can override all the parameters you want.
+Here is an example of the `./my_awesome_configs/exp/ext_sota.yaml` config file:
+
+```yaml
+# @package _global_
+
+defaults:
+  - override /algo: ext_sota
+  - override /env: atari
+  # select the model manager configs
+  - override /model_manager: ext_sota
+  - _self_
+
+algo:
+  total_steps: 65536
+  per_rank_batch_size: 64
+
+buffer:
+  share_data: False
+
+# override environment id
+env:
+  env:
+    id: MsPacmanNoFrameskip-v4
+
+# select which metrics to log
+metric:
+  aggregator:
+    metrics:
+      Loss/loss1:
+        _target_: torchmetrics.MeanMetric
+        sync_on_compute: ${metric.sync_on_compute}
+      Loss/loss2:
+        _target_: torchmetrics.MeanMetric
+        sync_on_compute: ${metric.sync_on_compute}
+```
+
+With `override /algo: ext_sota` in `defaults` you are specifying you want to use the new `ext_sota` algorithm, whereas, with `override /env: atari` you are specifying that you want to train your agent on an *Atari* environment.
+
+## Register the algorithm and the configs
+
+To let the `register_algorithm` decorator add our new `ext_sota` algorithm to the available algorithms registry we need first to create a new file called for example `my_awesome_main.py` in the root of the project:
+
+```bash
+.
+├── my_awesome_algo
+|   ├── __init__.py
+│   ├── agent.py
+│   ├── loss.py
+│   ├── ext_sota.py
+│   └── utils.py
+├── my_awesome_configs
+|   ├── algo
+|   |   └── ext_sota.yaml
+|   ├── exp
+|   |   └── ext_sota.yaml
+|   └── model_manager
+|       └── ext_sota.yaml
+└── my_awesome_main.py
+```
+
+containing the following:
+
+```python
+# my_awesome_main.py
+
+# This will trigger the algorithm registration of SheepRL
+from my_awesome_algo import ext_sota  # noqa: F401
+
+if __name__ == "__main__":
+    from sheeprl.cli import run
+
+    run()
+```
+
+While to let SheepRL know about the new configs we need to add a new file called `.env` to the root of the project containing the following env variable:
+
+```bash
+SHEEPRL_SEARCH_PATH=file://my_awesome_configs;pkg://sheeprl.configs
+```
+
+This tells SheepRL to search for configs in the `my_awesome_configs` folder and in the `configs` folder under the installed `sheeprl` package.
+
+## Run the experiment
+
+Then you can run your experiment with `python my_awesome_main.py exp=ext_sota env=... env.id=...`.
```

Comment:
At the beginning you wrote:

> Suppose that we have installed SheepRL through pip with `pip install sheeprl[box2d,dev,test]` and we want to add a new SoTA algorithm called `ext_sota` so that we can train an agent simply with `sheeprl exp=ext_sota env=... env.id=...` (without directly adding the new algorithm to the SheepRL codebase: i.e., there's no need to clone the repo locally).

Here you say that it has to write the `my_awesome_main.py` script. Can you run the `ext_sota` algorithm with the `sheeprl exp=ext_sota env=... env.id=...` command? Or you must implement the `my_awesome_main.py` script?

I think it is important to make this clear to the user so that he does not end up with errors that he struggles to resolve.

Reply:
Yeah, you're right! The user must implement the `my_awesome_main.py` script in order to run an experiment with that newly-added algorithm

Reply:
Great!